### ISSUE TITLE
[souncloud] -f bestaudio does not download the best quality available

### ISSUE THREAD

**Issue by lolcatw** [yt-dlp/yt-dlp#11008]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Canada

### Provide a description that is worded well enough to be understood

I run the following command to download a song

```sh
yt-dlp --cookies-from-browser firefox -f bestaudio https://soundcloud.com/111loggedin/c29uz3mgzm9yigdsb29texmgzgf5cya6mw
```
But it gives me a 64kbps opus file.

When I list out the available formats, I get this
```
ID           EXT  RESOLUTION │  FILESIZE  TBR PROTO │ VCODEC     ACODEC  ABR
────────────────────────────────────────────────────────────────────────────
hls_mp3_128  mp3  audio only │ ~27.02MiB 128k m3u8  │ audio only mp3    128k
http_mp3_128 mp3  audio only │ ~27.02MiB 128k http  │ audio only mp3    128k
hls_opus_64  opus audio only │ ~13.51MiB  64k m3u8  │ audio only opus    64k
```

In this instance, the hls or http streams are far superior. Why is it download the last one?

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
will@fuckedmachine:~$ yt-dlp --cookies-from-browser firefox -vU https://soundcloud.com/111loggedin/c29uz3mgzm9yigdsb29texmgzgf5cya6mw
[debug] Command-line config: ['--cookies-from-browser', 'firefox', '-vU', 'https://soundcloud.com/111loggedin/c29uz3mgzm9yigdsb29texmgzgf5cya6mw']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (pip)
[debug] Python 3.10.11 (CPython x86_64 64bit) - Linux-6.1.0-25-amd64-x86_64-with-glibc2.36 (OpenSSL 1.1.1n  15 Mar 2022, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.6-0 (setts), ffprobe 5.1.6-0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.18.0, brotli-1.0.9, certifi-2023.05.07, mutagen-1.47.0, requests-2.32.3, sqlite3-3.40.1, urllib3-2.0.2, websockets-12.0
[debug] Proxy map: {}
Extracting cookies from firefox
[debug] Extracting cookies from: "/home/will/.mozilla/firefox/nollfob0.default-release/cookies.sqlite"
Extracted 1785 cookies from firefox
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1830 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
[debug] Loading soundcloud.client_id from cache
[soundcloud] Verifying login token...
[soundcloud] Logging in
[soundcloud] Extracting URL: https://soundcloud.com/111loggedin/c29uz3mgzm9yigdsb29texmgzgf5cya6mw
[soundcloud] 111loggedin/c29uz3mgzm9yigdsb29texmgzgf5cya6mw: Downloading info JSON
[soundcloud] 1772862330: Downloading hls_mp3 format info JSON
[soundcloud] 1772862330: Downloading http_mp3 format info JSON
[soundcloud] 1772862330: Downloading hls_opus format info JSON
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 1772862330: Downloading 1 format(s): hls_opus_64
[debug] Invoking hlsnative downloader on "https://cf-hls-opus-media.sndcdn.com/playlist/5514313b-40e1-4b91-b779-6d3494c6d944.64.opus/playlist.m3u8?Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiKjovL2NmLWhscy1vcHVzLW1lZGlhLnNuZGNkbi5jb20vcGxheWxpc3QvNTUxNDMxM2ItNDBlMS00YjkxLWI3NzktNmQzNDk0YzZkOTQ0LjY0Lm9wdXMvcGxheWxpc3QubTN1OCoiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE3MjYzODI3MjB9fX1dfQ__&Signature=arKOsyTsRuCkJ-yLQT6BJnElHy9pbHzzaCjFnF3BPTF5os2MKZQy3hYBA37Qc7QlKICJK8mi4-XCbQOOfX-Ww~dwPZouj3jiuQnWOLOLATirW3YSToa~aKFSlMg75fcFxJeXUY3LOfp56x8~EYzhbzR~WzSa-DoSt1qATFNej6xLelUod~Vf3XrkdHBJTxnlayuemFyHqvdoG2pz2offjHDdFmOWdJ492PuyfWopCeMV~ZXFl0xAiveAA2fdCATWyxy9x6i8-RIWlosWw5yfLmlsTqKJxshDFstatkxsoRhkePd~2qedObXA5HXo78E51aMuoGAD7kMC-xGWCOeeIQ__&Key-Pair-Id=APKAI6TU7MMXM5DG6EPQ"
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 180
[download] Destination: c29uZ3MgZm9yIGdsb29teXMgZGF5cyA6Mw== [1772862330].opus
[download]   7.2% of ~  12.75MiB at  622.54KiB/s ETA 00:16 (frag 14/180)^C
ERROR: Interrupted by user
```

**Comment by lolcatw** [comment]
Also, it doesn't download the `download` quality when available, which is often the original file in `wav` format.

**Comment by rdamas** [comment]
Superior only bitrate-wise. An opus codec usually has higher quality than mp3 - audio codecs have made significant progress since mp3 came to birth.

**Comment by lolcatw** [comment]
Opus codec introduces coil whine noise with certain instruments and the highs are all scuffed. Mp3 tends to make things sound muddy and more bassy, which is much more tolerable. Their web interface loads up the 128K MP3 version as well, which indicates me that it is indeed the best quality.

**Comment by seproDev** [comment]
By default, yt-dlp prefers more modern codecs. If you want to prefer mp3 instead, use `-S acodec:mp3`

**Comment by lolcatw** [comment]
My bad, `download` didn't appear as I didn't pass firefox cookies.

**Comment by JunkiEDM** [comment]
I think this should be re-opened (ignoring the `download` format mistake) since, although its correct that opus is better than mp3, 64k opus isn't better than 128k mp3 (for music, 96k opus>128k mp3>64k opus imo).

Someone else made a detailed write-up of why 128k mp3 > 64k opus for music back in 2017 when SoundCloud decided to switch the default codec from 128k mp3 to 64k opus, which notes that many artists noticed the reduced audio quality from the change: https://opus-tests.neocities.org/backup
Soundcloud since reverted course, hence the "backup" path, and 128k mp3 is once again the default for the site, further supporting this.

Anecdotally I remember yt-dlp *used to* prefer 128k mp3 over 64k opus on SoundCloud even when these changes were made, I only noticed it preferring 64k opus a few months ago when an alias I made to yt-dlp for downloading from SoundCloud started downloading in opus rather than mp3.
